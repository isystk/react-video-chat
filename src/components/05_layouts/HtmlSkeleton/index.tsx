import React, { Children, FC, useState } from 'react'
import { connect } from '@/components/hoc'
import { ContainerProps, WithChildren } from 'types'
import useAppRoot from '@/stores/useAppRoot'
import Header from '@/components/04_organisms/Header'
import SideMenu from '@/components/04_organisms/SideMenu'
import Title, { TitleProps } from './Title'
import Head from 'next/head'
import { isReactElement } from '@/utils/general/object'
import { CssBaseline } from '@material-ui/core'

/** HtmlSkeleton Props */
export type HtmlSkeletonProps = WithChildren
/** Presenter Props */
export type PresenterProps = HtmlSkeletonProps & {
  title?: TitleProps['children']
  description: string
}

/** Presenter Component */
const HtmlSkeletonPresenter: FC<PresenterProps> = ({
  main,
  title,
  description,
  isMenuOpen,
  setMenuOpen,
  children,
  ...props
}) => (
  <>
    <Head>
      {/* タイトル */}
      <title>{title}</title>
      {/* favicon */}
      <link rel="icon" href="/favicon.ico" />
      {/* ブラウザテーマカラー */}
      <meta name="theme-color" content="#ffffff" />
      {/* サイト概要 */}
      <meta name="description" content={description} />
      {/* OGP 画像URL */}
      <meta property="og:image" content={'/ogp-image.png'} />
      {/* OGP タイトル */}
      <meta name="og:title" content={title} />
      {/* OGP サイト概要 */}
      <meta name="og:description" content={description} />
      {/* OGP Twitterカード */}
      <meta name="twitter:card" content="summary" />
      {/* apple ポータブル端末 アイコン */}
      <link rel="apple-touch-icon" href="/apple-touch-icon.png" />
      {/* manifest.json */}
      <link rel="manifest" href="/manifest.json" />
    </Head>
    <Context.Provider value={main}>
      <CssBaseline />
      <Header isMenuOpen={isMenuOpen} setMenuOpen={setMenuOpen} />
      <div>{children}</div>
      <SideMenu isMenuOpen={isMenuOpen} setMenuOpen={setMenuOpen} />
    </Context.Provider>
  </>
)

/** Container Component */
const HtmlSkeletonContainer: React.FC<
  ContainerProps<HtmlSkeletonProps, PresenterProps>
> = ({ presenter, children, ...props }) => {
  let title: TitleProps['children'] | undefined = undefined
  const main = useAppRoot()
  const [isMenuOpen, setMenuOpen] = useState(false)

  if (!main) return <></>

  const APP_NAME = process.env.APP_NAME

  children = Children.map(children, (child) =>
    isReactElement(child) && child.type === Title
      ? (title = `${child.props.children} | ${APP_NAME}`) && undefined
      : React.cloneElement(child)
  )
  if (!title) title = process.env.APP_NAME

  const description = `${APP_NAME} is an application that allows you to casually chat with your friends!`
  return presenter({
    main,
    title,
    description,
    isMenuOpen,
    setMenuOpen,
    children,
    ...props,
  })
}

export const Context = React.createContext(null)

export default connect<HtmlSkeletonProps, PresenterProps>(
  'HtmlSkeleton',
  HtmlSkeletonPresenter,
  HtmlSkeletonContainer
)

// Sub Component
export type { TitleProps }
export { Title }
