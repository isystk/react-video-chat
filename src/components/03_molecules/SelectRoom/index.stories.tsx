import { Meta, Story } from '@storybook/react'
import React from 'react'
import SelectRoom from './index'
import MainService from '@/services/main'
import { Context } from '@/components/05_layouts/HtmlSkeleton'

export default {
  title: '03_molecules/SelectRoom',
  component: SelectRoom,
} as Meta

const Template: Story = (props) => {
  const main = new MainService(() => ({}))
  main.setName('isystk')

  return (
    <Context.Provider value={main}>
      <SelectRoom />
    </Context.Provider>
  )
}

export const Primary = Template.bind({})
Primary.storyName = 'プライマリ'
Primary.args = {}
