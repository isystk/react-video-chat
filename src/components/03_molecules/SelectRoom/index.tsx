import React, { FC, useContext, useEffect, useState } from 'react'
import {
  Button,
  Grid,
  Paper,
  Table,
  TableBody,
  TableCell,
  TableContainer,
  TableHead,
  TableRow,
  Container,
} from '@material-ui/core'
import Pagination from '@material-ui/lab/Pagination'
import { ContainerProps, WithChildren } from 'types'
import { useStyles } from './styles'
import { connect } from '@/components/hoc'
import MainService from '@/services/main'
import Box from '@/components/01_atoms/Box'
import { Context } from '@/components/05_layouts/HtmlSkeleton'
import AddCircleIcon from '@material-ui/icons/AddCircle'
import EditIcon from '@material-ui/icons/Edit'
import DeleteIcon from '@material-ui/icons/Delete'
import { Room } from '@/services/room'
import * as _ from 'lodash'
import RoomRegistModal from '@/components/04_organisms/RoomRegistModal'

/** SelectRoomProps Props */
export type SelectRoomProps = WithChildren
/** Presenter Props */
export type PresenterProps = SelectRoomProps & {
  main
  classes
  page
  setPage
  pageCount
  allItems
  displayNum
  displayedItems
  setDisplayedItems
}

/** Presenter Component */
const SelectRoomPresenter: FC<PresenterProps> = ({
  main,
  classes,
  page,
  setPage,
  pageCount,
  allItems,
  displayNum,
  displayedItems,
  setDisplayedItems,
  openRoomRegistModal,
  selectRoom,
  setSelectRoom,
  ...props
}) => (
  <>
    <Container component="main">
      <Box title="部屋の選択">
        <Grid container style={{ padding: '20px' }}>
          <Grid item xs={12} style={{ marginBottom: '20px' }}>
            <Grid container justifyContent="space-between">
              <Grid item></Grid>
              <Grid item>
                <Button
                  color="primary"
                  type="button"
                  variant="contained"
                  startIcon={<AddCircleIcon />}
                  onClick={() => {
                    setSelectRoom(null)
                    openRoomRegistModal()
                  }}
                >
                  新規登録
                </Button>
              </Grid>
            </Grid>
          </Grid>
          <Grid item xs={12}>
            <TableContainer component={Paper} style={{ marginBottom: 30 }}>
              {/* componentにライブラリのPaperをつけることで立体感がでてよくなります */}
              <Table>
                <TableHead>
                  <TableRow style={{ backgroundColor: '#F2F2F2' }}>
                    <TableCell align="center">名称</TableCell>
                    <TableCell align="center">説明</TableCell>
                    <TableCell align="center">作成日</TableCell>
                    <TableCell align="center"></TableCell>
                  </TableRow>
                </TableHead>
                <TableBody>
                  {_.map(displayedItems, (e: Room) => (
                    <TableRow key={e.id}>
                      <TableCell>{e.name}</TableCell>
                      <TableCell>{e.description}</TableCell>
                      <TableCell align="center">{e.createdAt}</TableCell>
                      <TableCell align="center">
                        <Grid container spacing={1}>
                          <Grid item xs={12}>
                            <Button
                              variant="contained"
                              color="primary"
                              type="submit"
                              startIcon={<EditIcon />}
                              onClick={() => {
                                setSelectRoom(e)
                                openRoomRegistModal()
                              }}
                            >
                              変更
                            </Button>
                          </Grid>
                          <Grid item xs={12}>
                            <Button
                              variant="contained"
                              color="secondary"
                              type="submit"
                              startIcon={<DeleteIcon />}
                              onClick={async () => {
                                if (confirm('削除します。よろしいですか？'))
                                  console.log('delete')
                              }}
                            >
                              削除
                            </Button>
                          </Grid>
                        </Grid>
                      </TableCell>
                    </TableRow>
                  ))}
                </TableBody>
              </Table>
            </TableContainer>
            <Grid item xs={12}>
              <Pagination
                count={pageCount}
                page={page}
                variant="outlined"
                color="primary"
                size="small"
                onChange={(_event, index) => {
                  //ページ移動時にページ番号を更新
                  setPage(index)
                  //ページ移動時に表示データを書き換える
                  setDisplayedItems(
                    allItems.slice((index - 1) * displayNum, index * displayNum)
                  )
                }}
              />
            </Grid>
          </Grid>
        </Grid>
      </Box>
    </Container>
    <RoomRegistModal room={selectRoom} />
  </>
)

/** Container Component */
const SelectRoomContainer: React.FC<
  ContainerProps<SelectRoomProps, PresenterProps>
> = ({ presenter, children, ...props }) => {
  const main = useContext<MainService | null>(Context)
  if (!main) return <></>
  const classes = useStyles()

  const [page, setPage] = useState(1) //ページ番号
  const [pageCount, setPageCount] = useState<number>() //ページ数
  const [allItems, setAllItems] = useState<Room[]>([]) //全データ
  const [displayedItems, setDisplayedItems] = useState<Room[]>([]) //表示データ
  const displayNum = 3 //1ページあたりの項目数

  const [selectRoom, setSelectRoom] = useState<Room | null>(null)

  useEffect(() => {
    main.room.readRooms()
  }, [])

  useEffect(() => {
    if (main.room.rooms.length === 0) return
    setAllItems(main.room.rooms)
    //ページカウントの計算（今回は3項目/ページなので4ページ）
    setPageCount(Math.ceil(main.room.rooms.length / displayNum))
    //表示データを抽出
    setDisplayedItems(
      main.room.rooms.slice((page - 1) * displayNum, page * displayNum)
    )
  }, [main.room.rooms])

  if (main.self.name === '') return <></>
  if (main.room.name !== '') return <></>

  const openRoomRegistModal = () => {
    main.room.isOpen = true
    main.setAppRoot()
  }

  return presenter({
    children,
    main,
    classes,
    page,
    setPage,
    pageCount,
    allItems,
    displayNum,
    displayedItems,
    setDisplayedItems,
    openRoomRegistModal,
    selectRoom,
    setSelectRoom,
    ...props,
  })
}

export default connect<SelectRoomProps, PresenterProps>(
  'SelectRoom',
  SelectRoomPresenter,
  SelectRoomContainer
)
