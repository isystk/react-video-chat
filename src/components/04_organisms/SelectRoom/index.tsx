import React, { FC, useContext, useEffect, useState } from 'react'
import { Button, Grid, Container } from '@material-ui/core'
import { ContainerProps, WithChildren } from 'types'
import { useStyles } from './styles'
import { connect } from '@/components/hoc'
import MainService from '@/services/main'
import Box from '@/components/01_atoms/Box'
import { Context } from '@/components/05_layouts/HtmlSkeleton'
import RoomList from '@/components/03_molecules/RoomList'
import AddCircleIcon from '@material-ui/icons/AddCircle'
import { Room } from '@/services/room'
import RoomRegistModal from '@/components/04_organisms/RoomRegistModal'
import TextField from '@material-ui/core/TextField'

/** SelectRoomProps Props */
export type SelectRoomProps = WithChildren
/** Presenter Props */
export type PresenterProps = SelectRoomProps & {
  main
  classes
  openRoomRegistModal
  selectRoom
  setSelectRoom
}

/** Presenter Component */
const SelectRoomPresenter: FC<PresenterProps> = ({
  main,
  classes,
  openRoomRegistModal,
  selectRoom,
  filerName,
  onChangeFilterName,
  ...props
}) => (
  <>
    <Container component="main">
      <Box title="部屋の選択">
        <Grid container style={{ padding: '20px' }}>
          <Grid item xs={12} style={{ marginBottom: '20px' }}>
            <Grid container justifyContent="space-between">
              <Grid item>
                <TextField
                  fullWidth
                  size="small"
                  label="部屋名"
                  name="roomName"
                  onChange={onChangeFilterName}
                  value={filerName}
                  variant="outlined"
                />
              </Grid>
              <Grid item>
                <Button
                  color="primary"
                  type="button"
                  variant="contained"
                  startIcon={<AddCircleIcon />}
                  onClick={() => {
                    openRoomRegistModal(null)
                  }}
                >
                  新規登録
                </Button>
              </Grid>
            </Grid>
          </Grid>
          <Grid item xs={12}>
            <RoomList openRoomRegistModal={openRoomRegistModal} />
          </Grid>
        </Grid>
      </Box>
    </Container>
    <RoomRegistModal room={selectRoom} />
  </>
)

/** Container Component */
const SelectRoomContainer: React.FC<
  ContainerProps<SelectRoomProps, PresenterProps>
> = ({ presenter, children, ...props }) => {
  const main = useContext<MainService | null>(Context)
  if (!main) return <></>
  const classes = useStyles()
  const [selectRoom, setSelectRoom] = useState<Room | null>(null)
  const [filerName, setFilterName] = useState<string>('')

  if (main.self.name === '') return <></>
  if (main.room.name !== '') return <></>

  const openRoomRegistModal = (room) => {
    main.room.isOpen = true
    setSelectRoom(room)
    main.setAppRoot()
  }

  const onChangeFilterName = (e) => {
    const name = e.target.value
    setFilterName(name)
    main.room.readRooms(name)
  }

  return presenter({
    children,
    main,
    classes,
    openRoomRegistModal,
    selectRoom,
    setSelectRoom,
    filerName,
    onChangeFilterName,
    ...props,
  })
}

export default connect<SelectRoomProps, PresenterProps>(
  'SelectRoom',
  SelectRoomPresenter,
  SelectRoomContainer
)
