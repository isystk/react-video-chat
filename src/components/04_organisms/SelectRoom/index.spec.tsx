import React from 'react'
import renderer from 'react-test-renderer'
import SelectRoom from './index'
import { Context } from '@/components/05_layouts/HtmlSkeleton'
import '@testing-library/jest-dom/extend-expect'
import MainService from '@/services/main'

describe('SelectRoom', () => {
  it('Match Snapshot', () => {
    const main = new MainService(() => ({}))
    main.setName('isystk')

    const component = renderer.create(
      <Context.Provider value={main}>
        <SelectRoom />
      </Context.Provider>
    )
    const tree = component.toJSON()

    expect(tree).toMatchSnapshot()
  })
})
