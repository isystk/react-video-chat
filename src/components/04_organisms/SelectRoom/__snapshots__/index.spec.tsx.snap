// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SelectRoom Match Snapshot 1`] = `
<main
  className="MuiContainer-root MuiContainer-maxWidthLg css-1oqqzyl-MuiContainer-root"
>
  <nav
    aria-label="breadcrumb"
    className="MuiTypography-root MuiTypography-body1 MuiBreadcrumbs-root css-1kynpjt css-63vwl1-MuiTypography-root-MuiBreadcrumbs-root"
  >
    <ol
      className="MuiBreadcrumbs-ol css-4pdmu4-MuiBreadcrumbs-ol"
    >
      <li
        className="MuiBreadcrumbs-li"
      >
        <a
          color="inherit"
          href="/"
          onClick={[Function]}
          onMouseEnter={[Function]}
          onTouchStart={[Function]}
        >
          TOP
        </a>
      </li>
      <li
        aria-hidden={true}
        className="MuiBreadcrumbs-separator css-1wuw8dw-MuiBreadcrumbs-separator"
      >
        /
      </li>
      <li
        className="MuiBreadcrumbs-li"
      >
        <span
          className="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
        >
          部屋の選択
        </span>
      </li>
    </ol>
  </nav>
  <div
    className="css-heco0o MuiBox-root css-0"
  >
    <div
      className="MuiTypography-root MuiTypography-body1 css-13emhng css-vkc4v8-MuiTypography-root"
    >
      部屋の選択
    </div>
    <div
      className="MuiGrid-root MuiGrid-container css-11lq3yg-MuiGrid-root"
      style={
        Object {
          "padding": "20px",
        }
      }
    >
      <div
        className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-1idn90j-MuiGrid-root"
        style={
          Object {
            "marginBottom": "20px",
          }
        }
      >
        <div
          className="MuiGrid-root MuiGrid-container css-1xzqqmo-MuiGrid-root"
        >
          <div
            className="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
          >
            <div
              className="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-wb57ya-MuiFormControl-root-MuiTextField-root"
            >
              <label
                className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeSmall MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeSmall MuiInputLabel-outlined css-1pysi21-MuiFormLabel-root-MuiInputLabel-root"
                data-shrink={false}
                htmlFor=":r0:"
                id=":r0:-label"
              >
                部屋名
              </label>
              <div
                className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-sizeSmall css-md26zr-MuiInputBase-root-MuiOutlinedInput-root"
                onClick={[Function]}
              >
                <input
                  aria-invalid={false}
                  autoFocus={false}
                  className="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputSizeSmall css-1n4twyu-MuiInputBase-input-MuiOutlinedInput-input"
                  disabled={false}
                  id=":r0:"
                  name="roomName"
                  onAnimationStart={[Function]}
                  onBlur={[Function]}
                  onChange={[Function]}
                  onFocus={[Function]}
                  required={false}
                  type="text"
                  value=""
                />
                <fieldset
                  aria-hidden={true}
                  className="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
                >
                  <legend
                    className="css-yjsfm1"
                  >
                    <span>
                      部屋名
                    </span>
                  </legend>
                </fieldset>
              </div>
            </div>
          </div>
          <div
            className="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
          >
            <button
              className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium css-sghohy-MuiButtonBase-root-MuiButton-root"
              disabled={false}
              onBlur={[Function]}
              onClick={[Function]}
              onContextMenu={[Function]}
              onDragLeave={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              tabIndex={0}
              type="button"
            >
              <span
                className="MuiButton-startIcon MuiButton-iconSizeMedium css-1d6wzja-MuiButton-startIcon"
              >
                <svg
                  aria-hidden={true}
                  className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                  data-testid="AddCircleIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm5 11h-4v4h-2v-4H7v-2h4V7h2v4h4v2z"
                  />
                </svg>
              </span>
              新規登録
            </button>
          </div>
        </div>
      </div>
      <div
        className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-1idn90j-MuiGrid-root"
      >
        <div
          className="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 css-1fw5qro-MuiPaper-root"
        >
          <div
            className="MuiTableContainer-root css-rorn0c-MuiTableContainer-root"
            style={
              Object {
                "maxHeight": 485,
              }
            }
          >
            <table
              aria-label="sticky table"
              className="MuiTable-root MuiTable-stickyHeader css-xn82ks-MuiTable-root"
              role={null}
            >
              <thead
                className="MuiTableHead-root css-15wwp11-MuiTableHead-root"
                role={null}
              >
                <tr
                  className="MuiTableRow-root MuiTableRow-head css-1q1u3t4-MuiTableRow-root"
                  role={null}
                  style={
                    Object {
                      "backgroundColor": "#F2F2F2",
                    }
                  }
                >
                  <th
                    aria-sort={null}
                    className="MuiTableCell-root MuiTableCell-head MuiTableCell-stickyHeader MuiTableCell-alignCenter MuiTableCell-sizeMedium css-dwuj3p-MuiTableCell-root"
                    scope="col"
                    style={
                      Object {
                        "width": 300,
                      }
                    }
                  >
                    名称
                  </th>
                  <th
                    aria-sort={null}
                    className="MuiTableCell-root MuiTableCell-head MuiTableCell-stickyHeader MuiTableCell-alignCenter MuiTableCell-sizeMedium css-dwuj3p-MuiTableCell-root"
                    scope="col"
                    style={
                      Object {
                        "width": 500,
                      }
                    }
                  >
                    説明
                  </th>
                  <th
                    aria-sort={null}
                    className="MuiTableCell-root MuiTableCell-head MuiTableCell-stickyHeader MuiTableCell-alignCenter MuiTableCell-sizeMedium css-dwuj3p-MuiTableCell-root"
                    scope="col"
                    style={
                      Object {
                        "width": 100,
                      }
                    }
                  >
                    作成日
                  </th>
                  <th
                    aria-sort={null}
                    className="MuiTableCell-root MuiTableCell-head MuiTableCell-stickyHeader MuiTableCell-alignCenter MuiTableCell-sizeMedium css-dwuj3p-MuiTableCell-root"
                    scope="col"
                  />
                </tr>
              </thead>
              <tbody
                className="MuiTableBody-root css-apqrd9-MuiTableBody-root"
                role={null}
              />
            </table>
          </div>
          <div
            className="MuiTablePagination-root css-jtlhu6-MuiTablePagination-root"
          >
            <div
              className="MuiToolbar-root MuiToolbar-gutters MuiToolbar-regular MuiTablePagination-toolbar css-78c6dr-MuiToolbar-root-MuiTablePagination-toolbar"
            >
              <div
                className="MuiTablePagination-spacer css-1psng7p-MuiTablePagination-spacer"
              />
              <p
                className="MuiTablePagination-selectLabel css-pdct74-MuiTablePagination-selectLabel"
                id=":r2:"
              >
                Rows per page:
              </p>
              <div
                className="MuiInputBase-root MuiInputBase-colorPrimary css-16c50h-MuiInputBase-root-MuiTablePagination-select"
                onClick={[Function]}
              >
                <div
                  aria-expanded="false"
                  aria-haspopup="listbox"
                  aria-labelledby=":r2: :r1:"
                  className="MuiSelect-select MuiTablePagination-select MuiSelect-standard MuiInputBase-input css-194a1fa-MuiSelect-select-MuiInputBase-input"
                  id=":r1:"
                  onBlur={[Function]}
                  onFocus={[Function]}
                  onKeyDown={[Function]}
                  onMouseDown={[Function]}
                  role="button"
                  tabIndex={0}
                >
                  3
                </div>
                <input
                  aria-hidden={true}
                  className="MuiSelect-nativeInput css-yf8vq0-MuiSelect-nativeInput"
                  onAnimationStart={[Function]}
                  onChange={[Function]}
                  tabIndex={-1}
                  value={3}
                />
                <svg
                  aria-hidden={true}
                  className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiSelect-icon MuiTablePagination-selectIcon MuiSelect-iconStandard css-pqjvzy-MuiSvgIcon-root-MuiSelect-icon"
                  data-testid="ArrowDropDownIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M7 10l5 5 5-5z"
                  />
                </svg>
              </div>
              <p
                className="MuiTablePagination-displayedRows css-levciy-MuiTablePagination-displayedRows"
              >
                0–0 of 0
              </p>
              <div
                className="MuiTablePagination-actions"
              >
                <button
                  aria-label="Go to previous page"
                  className="MuiButtonBase-root Mui-disabled MuiIconButton-root Mui-disabled MuiIconButton-colorInherit MuiIconButton-sizeMedium css-zylse7-MuiButtonBase-root-MuiIconButton-root"
                  disabled={true}
                  onBlur={[Function]}
                  onClick={[Function]}
                  onContextMenu={[Function]}
                  onDragLeave={[Function]}
                  onFocus={[Function]}
                  onKeyDown={[Function]}
                  onKeyUp={[Function]}
                  onMouseDown={[Function]}
                  onMouseLeave={[Function]}
                  onMouseUp={[Function]}
                  onTouchEnd={[Function]}
                  onTouchMove={[Function]}
                  onTouchStart={[Function]}
                  tabIndex={-1}
                  title="Go to previous page"
                  type="button"
                >
                  <svg
                    aria-hidden={true}
                    className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                    data-testid="KeyboardArrowLeftIcon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M15.41 16.09l-4.58-4.59 4.58-4.59L14 5.5l-6 6 6 6z"
                    />
                  </svg>
                </button>
                <button
                  aria-label="Go to next page"
                  className="MuiButtonBase-root Mui-disabled MuiIconButton-root Mui-disabled MuiIconButton-colorInherit MuiIconButton-sizeMedium css-zylse7-MuiButtonBase-root-MuiIconButton-root"
                  disabled={true}
                  onBlur={[Function]}
                  onClick={[Function]}
                  onContextMenu={[Function]}
                  onDragLeave={[Function]}
                  onFocus={[Function]}
                  onKeyDown={[Function]}
                  onKeyUp={[Function]}
                  onMouseDown={[Function]}
                  onMouseLeave={[Function]}
                  onMouseUp={[Function]}
                  onTouchEnd={[Function]}
                  onTouchMove={[Function]}
                  onTouchStart={[Function]}
                  tabIndex={-1}
                  title="Go to next page"
                  type="button"
                >
                  <svg
                    aria-hidden={true}
                    className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                    data-testid="KeyboardArrowRightIcon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M8.59 16.34l4.58-4.59-4.58-4.59L10 5.75l6 6-6 6z"
                    />
                  </svg>
                </button>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</main>
`;
